src/NextGen/Framework/FormTemplates/Registrars/FormTemplateRegistrar.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Give\NextGen\Framework\FormTemplates\Registrars;
+
+use Give\Framework\Exceptions\Primitives\InvalidArgumentException;
+use Give\Framework\PaymentGateways\Exceptions\OverflowException;
+use Give\Log\Log;
+use Give\NextGen\Framework\FormTemplates\FormTemplate;
+
+/**
+ * @unreleased
+ */
+class FormTemplateRegistrar
+{
+    protected $templates = [];
+
+    /**
+     * @unreleased
+     */
+    public function getTemplates(): array
+    {
+        return $this->templates;
+    }
+
+    /**
+     * @unreleased
+     *
+     * @throws InvalidArgumentException
+     */
+    public function getTemplate(string $id): FormTemplate
+    {
+        if (!$this->hasTemplate($id)) {
```

Comment:
This check is redundent, as `give()` will throw this exception itself.

```
Give\Framework\Exceptions\Primitives\InvalidArgumentException with message 'Target class [Foo] does not exist.'
```

Instead, we could catch it and re-throw a more specific exception, like Form Template class does not exist.

Reply:
This is just checking if the ID/key exists within the registrar, not if the class exists.

Reply:
Ah, fair point.